File: challenger-backend/src/main/java/com/kameo/challenger/odb/ConfirmationLinkODB.java
package com.kameo.challenger.odb;

import com.kameo.challenger.odb.api.IIdentity;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;

/**
 * Created by kmyczkowska on 2016-09-02.
 */
@Entity
@Data
@ToString(of= IIdentity.id_column)
@NoArgsConstructor

public class ConfirmationLinkODB implements IIdentity {

    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    long id;
    String uid;

    @Enumerated
    ConfirmationLinkType confirmationLinkType;

    String fieldLogin;
    String fieldPasswordHash;
    String fieldSalt;
    String email;
    Long challengeId;


}

File: challenger-backend/src/main/java/com/kameo/challenger/odb/ConfirmationLinkODB.kt
package com.kameo.challenger.odb

import com.kameo.challenger.odb.api.IIdentity
import javax.persistence.Entity
import javax.persistence.Enumerated
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.validation.constraints.NotNull


@Entity
data class ConfirmationLinkODB(@Id
                               @GeneratedValue(strategy = javax.persistence.GenerationType.AUTO)
                               override var id: Long = 0):IIdentity {


    lateinit var uid: String;

    @Enumerated
    lateinit var confirmationLinkType: ConfirmationLinkType;

    var fieldLogin: String? = null
    var fieldPasswordHash: String? = null
    var fieldSalt: String? = null
    var email: String? = null
    var challengeId: Long? = null
}